lib/data/repository/marvel_api_provider.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:dio/dio.dart';
+
+import 'package:marvel/data/model/character.dart';
+import 'package:marvel/data/model/series.dart';
+
+class MarvelApiProvider {
```

Comment:
It is not really Provider it is more like Api or ApiClient. Provider usually provides one of multiple services based on different conditions calculated within the Provider.

Reply:
done